src/radical/entk/appman/wfprocessor.py

```diff
@@ -100,145 +110,173 @@ def _initialize_workflow(self):
-
-                                        if executable_stage not in scheduled_stages:
-                                            scheduled_stages.append(
-                                                executable_stage)
-
+                        # If Pipeline is in the final state or suspended, we
+                        # skip processing it.
+                        if pipe.state in states.FINAL or  \
+                           pipe.completed or \
+                           pipe.state == states.SUSPENDED:
+
+                            continue
+
+                        if pipe.state == states.INITIAL:
+
+                            # Set state of pipeline to SCHEDULING if it is
+                            # in INITIAL.
+                            local_transition(obj=pipe,
+                                             obj_type='Pipeline',
+                                             new_state=states.SCHEDULING,
+                                             profiler=self._prof,
+                                             logger=self._logger,
+                                             reporter=self._report)
+
+                        # Get the next stage of this pipeline to process
+                        exec_stage = pipe.stages[pipe.current_stage - 1]
+
+                        if not exec_stage.uid:
+                            # TODO: Move parent uid, name assignment to
+                            # assign_uid function
+                            exec_stage.parent_pipeline['uid'] = pipe.uid
+                            exec_stage.parent_pipeline['name'] = pipe.name
+                            exec_stage._assign_uid(self._sid)
+
+                        # We need to check both INITIAL and SCHEDULED as
+                        # there maybe a task in the stage that needs to be
+                        # resubmitted.
+                        exec_tasks = list()
+                        if exec_stage.state in [states.INITIAL,
+                                                states.SCHEDULED]:
+
+                            # If its a new stage, update its state
+                            if exec_stage.state == states.INITIAL:
+
+                                local_transition(obj=exec_stage,
+                                                    obj_type='Stage',
+                                                    new_state=states.SCHEDULING,
+                                                    profiler=self._prof,
+                                                    logger=self._logger,
+                                                    reporter=self._report)
+
+                            exec_tasks = exec_stage.tasks
+
+                        for exec_task in exec_tasks:
+
+                            if (exec_task.state == states.INITIAL) or \
+                                ((exec_task.state == states.FAILED) and
+                                    (self._resubmit_failed)):
+
```

Comment:
Some brackets above are not needed and only make (human) parsing more difficult.  I would suggest:
```
if exec_task.state == states.INITIAL or \
    (exec_task.state == states.FAILED and self._resubmit_failed):
```

But also, is `_resubmit_failed` only ever true if the task is `FAILED` in the first place, and thus implies that state already?  If not, shouldn't that be the case?

Reply:
_resubmit_failed is a flag that the user sets. So if the flag is set and the task state is FAILED, we have to resubmit the task. Does that answer your question?

Reply:
yes